Updated documentation to include --update-only flag

Gyan-max opened the pull request:
**Title:** Fix File Copy and Checkout Issues in PowerShell & Git
**Description:**
This PR addresses issues encountered while copying files and checking out specific files from the feature/update-only-flag branch.

**Issues Resolved:**
PowerShell Copy-Item Errors

Incorrect syntax while copying multiple files resulted in errors.
Missing files (CHECKERS.md and index.rst) caused the copy command to fail.
Git Checkout Command Not Recognized

checkout was mistakenly used instead of git checkout.
Files needed to be fetched from the correct branch before checkout.
**Fix Implemented:**
✅ Used the correct Copy-Item syntax with a comma-separated list.
✅ Verified file existence before copying to prevent errors.
✅ Used git checkout feature/update-only-flag -- <files> correctly.
✅ Ensured the correct branch was checked out and updated.

**Testing & Verification:**
 Verified files exist before copying.
 Successfully checked out files from feature/update-only-flag.
 Confirmed changes with git status and committed updates.

terriko commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,7 @@
+
+# CONTRIBUTING
+
+# CVE Binary Tool
```
```suggestion
```

I don't think we need extra header/title lines here.

terriko commented on README.md:
```diff
@@ -1,5 +1,10 @@
+
+# README
+
+# CVE Binary Tool
```
```suggestion
```
These titles/headers are also not needed.

terriko commented on doc/CHECKERS.md:
```diff
@@ -1 +1,2 @@
+# Checkers in CVE Binary Tool
```
Does adding this not break the way the link works when the files are parsed?  (Files with just a filename in them are effectively symbolic links)

terriko commented on CHECKERS.md:
```diff
@@ -0,0 +1 @@
+../cve_bin_tool/checkers/README.md
```
I don't think we want this file as a top level file when there's already a link for it within `doc/`

terriko commented on doc/index.rst:
```diff
@@ -16,18 +16,32 @@ The CVE Binary Tool helps you determine if your system includes known vulnerabil
    MANUAL.md
    CSV2CVE.md
    how_to_guides/index
-   sboms_for_humans/index
```
Why are the sboms for humans links being removed?  Are they broken?

terriko commented on index.rst:
```diff
@@ -0,0 +1,36 @@
+.. CVE Binary Tool documentation master file, created by
```
We already have an index.rst in doc/ and should not have a second one in the main directory, so this is not needed.

terriko reviewed (changes requested):
I've flagged a bunch of changes needed.  I'm not sure why so many things got moved into / -- we have to have some of our docs there because of github's expectations about doc files but anything not required to be in `/` should be in `doc/` please!

terriko commented on README.md:
```diff
@@ -1,5 +1,7 @@
+
```
```suggestion
```

not needed.

terriko commented on doc/index.rst:
```diff
@@ -16,18 +16,33 @@ The CVE Binary Tool helps you determine if your system includes known vulnerabil
    CSV2CVE.md
    how_to_guides/index
-   sboms_for_humans/index
+   sobms_for_humans/index
```
```suggestion
   sboms_for_humans/index
```
I think this is a typo?

Gyan-max commented:
"I've addressed all requested changes:
✅ Moved documentation files back to doc/
✅ Fixed the typo in doc/index.rst
✅ Removed unnecessary changes in README.md
✅ Ran linting and fixed all issues.
@terriko Please review again. Thank you!"

Gyan-max commented:
Hi @terriko 
Can you please guide me how can I solve this issue effectively because I'm getting little bit confused regarding the steps that I had to keep in my while resolving it.
ThankYou

terriko commented:
If you don't want to recreate the PR:
- remove each unneeded file using `git checkout origin/main --FILENAME`  -- in this case it's basically every .py file that got changed since this is supposed to be a docs update, so you'll be doing this 25 times or so.

If you do want to make a new PR, copy the changed files you care about somewhere for a  bit.   I think you probably want to keep your `.gitignore` and your changes should be in `doc/MANUAL.md` I'm going to pretend you're copying them to a directory named `~/tmp` below but you can adjust accordingly:

```
mkdir ~/tmp
cp .gitignore ~/tmp
cp doc/MANUAL.md ~/tmp
````

Once you have those stashed somewhere, go ahead and make a new completely clean branch based off `origin/main`.  Usually this is the part that people miss, so you wind up making a new branch based on whatever branch you happened to be on and wind up with a bunch of unrelated changes instead of a nice clean branch ready for merging with `origin/main`

```
git checkout origin/main
git branch -b update_only_doc_new
```

Then copy your changes into the new branch and git add them explicitly by filename:
```
cp ~/tmp/.gitignore .
cp ~/tmp/MANUAL.md doc/MANUAL.md
git add .gitignore doc/MANUAL.md
git commit --signoff
```

You should now have a new branch based on origin/main with only those changes!
